### PR TITLE
Added a note about API on TCP and restructured links

### PR DESCRIPTION
**Pull request by kayrus**
/cc @jonboulle @joshix @robszumski 

resolves https://github.com/coreos/fleet/issues/1414

### CODE REVIEW

File: Documentation/deployment-and-configuration.md [hunk 1]
```diff
@@ -40,24 +38,23 @@ fleet's API is served using systemd socket activation.
...
 
-After you've written the file, call `systemctl daemon-reload` to load the new drop-in, followed by `systemctl stop fleet.service; systemctl restart fleet.socket; systemctl start fleet.service`.
+After you've written the file, call `systemctl daemon-reload` to load the new [drop-in], followed by `systemctl stop fleet.service; systemctl restart fleet.socket; systemctl start fleet.service`.
 
 Once the socket is running, the fleet API will be available at `http://${ListenStream}/fleet/v1`, where `${ListenStream}` is the value of the `ListenStream` option used in your socket file.
 This endpoint is accessible directly using tools such as curl and wget, or you can use fleetctl like so: `fleetctl --endpoint http://${ListenStream} <command>`.
-For more information, see the [official API documentation][api-doc].
 
-[api-doc]: https://github.com/coreos/fleet/blob/master/Documentation/api-v1.md
+*NOTE:* It is not recommended to listen fleet API TCP socket over public and even private networks. Please use [ssh tunnel][ssh-tunnel] to access remote fleet API.
```
Comment (robszumski) [inline Documentation/deployment-and-configuration.md]: Can we rephrase this with out using "note:"? IMO these are the 90% use-cases when you say API, so this is definitely not just a note.

Can we explain _why_ this isn't recommended?

File: Documentation/deployment-and-configuration.md [hunk 2]
```diff
@@ -40,24 +38,23 @@ fleet's API is served using systemd socket activation.
...
+
+For more information, see the [official API documentation][api-doc].
```
Comment (robszumski) [inline Documentation/deployment-and-configuration.md]: There isn't any more information in here.

File: Documentation/using-the-client.md [hunk 3]
```diff
@@ -9,34 +9,48 @@ The `fleetctl` binary is included in all CoreOS distributions, so it is as simpl
...
+```
 
 Alternatively, `--endpoint` can be provided through the `FLEETCTL_ENDPOINT` environment variable:
 
-    FLEETCTL_ENDPOINT=http://<IP:[PORT]> fleetctl list-units
+```sh
+FLEETCTL_ENDPOINT=http://<IP:[PORT]> fleetctl list-units
+```
+
+*NOTE:* It is not recommended to listen fleet API TCP socket over public and even private networks. Please use [ssh tunnel][ssh-tunnel] to access remote fleet API.
```
Comment (robszumski) [inline Documentation/using-the-client.md]: same as https://github.com/coreos/fleet/pull/1421/files#r51611458

**Comment by robszumski** [comment]
What's the rational behind the port changes? For better readability, can you split this into separate commits next time?

**Comment by jonboulle** [comment]
Port reordering makes sense to me, favouring the newer IANA ports..

**Comment by jonboulle** [comment]
@kayrus can you please try follow our [commit guidelines](https://github.com/coreos/fleet/blob/master/CONTRIBUTING.md#format-of-the-commit-message)?

```
docs: add warning about API sockets, clean up links
```

**Comment by kayrus** [comment]
> can you please try follow our commit guidelines?

@jonboulle will do after review and squash

**Comment by kayrus** [comment]
@jonboulle squashed commits.

**Comment by joshix** [comment]
LGTM

**Pull request merged**